### ISSUE TITLE
DB schema/entity listing: order by DB insertion order (vs. by user-given timestamp)?

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#622]
My intuition and experience is that the default sort order of entities in system is often the order with which data was inserted into the database. 

When users use the API/ui to explore data, we by default sort most(all?) entities by the timestamp property that was after all user-given. 

Examples:

https://github.com/conbench/conbench/blob/2b1653f5582bf397ffaa10278062949a89a2ff8b/conbench/api/benchmarks.py#L151

```
            benchmark_results = BenchmarkResult.all(
                order_by=BenchmarkResult.timestamp.desc(), limit=500
            )
```

https://github.com/conbench/conbench/blob/2b1653f5582bf397ffaa10278062949a89a2ff8b/conbench/api/runs.py#L119

```
runs = Run.all(order_by=Run.timestamp.desc(), limit=1000)
```

----------

I think this is fine as long as those timestamps are not really error-prone.

I think in addition to this sort order we may want to offer a sort-by-db-insertion order. And in the long run maybe this should even be the default?

**Comment by austin3dickey** [comment]
This reminds me that it might be nice to have `created_at`, `created_by`, `updated_at`, and `updated_by` columns on our tables for auditing purposes. That might be helpful as we move toward asking more questions of users in the UI. We could then rely on the sort order of `created_at` if we wanted a view like this.

(Personally, on the homepage of the webapp, I like that BenchmarkResults are sorted by Run timestamp, but I don't really have an opinion on how the API returns results)

**Comment by jgehrcke** [comment]
> This reminds me that it might be nice to have created_at, created_by, updated_at, and updated_by columns on our tables for auditing purposes. 

Absolutely. Most more or less serious database-driven applications have this. 

> I like that BenchmarkResults are sorted by Run timestamp!

I agree! But with the above-mentioned caveat of "as long as those timestamps are not really error-prone". As these are user-given, funny things might happen that then still need to be discoverable. Yes, with reversed sort one might even find all those runs that supposedly ran on January 1st, 1970 :). Yeah, it's good to have some guarantees.
